### ISSUE TITLE
Higher level operations for neural networks

### ISSUE THREAD

**Issue by nkreeger** [PAIR-code/deeplearnjs#410]
The TF API has some nice wrappers for building simple neural networks. For example, `tf.contrib.layers.fully_connected() ` provides some simplicity for automatically providing weights, biases, and activation functions. This would make demos/setup much easier for simple training implementations w/o having to setup those variables manually and creating a dense graph-layer.

**Comment by nsthorat** [comment]
We started building higher level APIs, but we decided to put that on hold until the lower level APIs settled. Once we have eager, we can start to think about that.

Some links:
https://github.com/PAIR-code/deeplearnjs/blob/master/src/graph/graph.ts#L29

This was the first "layer" we built, which automatically constructed variables. Building the analogy once eager is done will be great :)

**Comment by dsmilkov** [comment]
With 0.5.0 out, we deprecated the graph api, which includes GraphLayers. We decided to not have higher level operations as part of the core dl.js library.
